src/test/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PandocTest.kt

```diff
@@ -15,44 +15,61 @@
  */
 package com.bardsoftware.backend.fleet.rmarkdown
 
+import com.bardsoftware.papeeria.backend.tex.CompileRequest
 import com.google.common.io.Files
-import com.typesafe.config.ConfigFactory
-import com.typesafe.config.ConfigValueFactory
+import com.google.pubsub.v1.PubsubMessage
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.typesafe.config.Config
 import org.junit.After
 import org.junit.Before
 import org.junit.Test
 import org.mockito.Mockito
+import org.mockito.Mockito.`when`
 import java.io.File
 import java.nio.file.Paths
-import kotlin.test.assertEquals
 import kotlin.test.assertTrue
 
 class PandocTest {
-    private val CP_COMMAND = "cp \${source} \${dest}"
+    private val CP_COMMAND = "cp \${projectRootAbsPath}/\${workingDirRelPath}/\${inputFileName} \${outputFileName}"
+    private val CONFIG_KEY = "pandoc.compile.command"
     private val tasksDir = "tasks"
+    private val taskId = "taskId"
 
     @Before
     fun createDir() {
-        File(this.tasksDir).mkdir()
+        val dir = File(this.tasksDir)
+        dir.mkdir()
+        dir.resolve(taskId).mkdir()
     }
 
     @Test
     fun basicCompile() {
         val source = Paths.get("src", "test", "resources", "example.Rmd").toString()
         val outputName = Files.getNameWithoutExtension(source) + ".tex"
-        val outputFile = Paths.get(tasksDir).resolve(outputName).toFile()
+        val outputFile = Paths.get(tasksDir).resolve(taskId).resolve(outputName).toFile()
 
-        val publisher = Mockito.mock(Publisher::class.java)
-        val markdownReceiver = MarkdownTaskReceiver(null, tasksDir, publisher)
+        val publisher = mock<Publisher> {
+            on { publish(any(), any()) }.then{}
+        }
 
-        val mockConfig = ConfigFactory
-                .empty()
-                .withValue("pandoc.compile.command", ConfigValueFactory.fromAnyRef(CP_COMMAND))
-        val cpArguments = mapOf("source" to source, "dest" to outputFile.toString())
-        val exitCode = markdownReceiver.convertMarkdown(cpArguments, mockConfig)
+        val mockConfig = Mockito.mock(Config::class.java)
```

Comment:
Any sense to replace this with mockito-kotlin stuff?

Reply:
Done